source/docs/hardware-reference/talonfx/improving-performance-with-current-limits.rst

```diff
@@ -0,0 +1,216 @@
+         :alt: Graph with no stator limit applied and a peak accel around 170 rotations/second²
+
+   .. grid-item-card:: With 80 A stator limit (~75 rotations/second²)
+
+      .. image:: images/with-stator-limit-accel.png
+         :alt: Graph with stator limit applied and a peak accel around 75 rotations/second²
+
+Supply Limits
+^^^^^^^^^^^^^
+
+.. important:: Supply limits are not functional in torque based control modes. Use the ``PeakForwardTorqueCurrent`` and ``PeakReverseTorqueCurrent`` configs instead.
+
+Supply current is the current drawn from the battery. As a result, limiting supply current can be useful to prevent fuses from resetting and breakers from tripping, as well as to improve the longevity of the battery.
+
+.. tip:: Since stator current limits also limit supply current, it is often not necessary to enable both limits.
+
+When the ``SupplyCurrentThreshold`` has elapsed for ``SupplyTimeThreshold`` amount of time, the supply current limiter will activate and reduce motor output until supply current is within range of the limit.
+
+In the rare case where the robot experiences brownouts despite configuring stator current limits, a supply current limit can also further help avoid brownouts. However, such brownouts are most commonly caused by a bad battery or poor power wiring, so those should be examined first.
+
+Determining Current Limits
+--------------------------
+
+While supply current limits can be theoretically estimated by calculating max supply draw for every mechanism, stator limits are not easy to estimate. Determine your stator current limits first, then supply only if necessary. In many cases, stator current limits are sufficient to prevent battery brownouts, as stator current limits also limit supply current.
+
+Preventing Wheel Slip
+^^^^^^^^^^^^^^^^^^^^^
+
+Stator current limits are excellent at preventing wheel slip, which is necessary to maximize power output of the system. To determine wheel slip, perform the following instructions:
+
+1. Place the robot on carpet against a wall.
+2. Begin plotting velocity and stator current in :doc:`Tuner X </docs/tuner/plotting>`.
+3. Slowly increase voltage output until the velocity becomes non-zero and the stator current drops.
+
+Set your stator current limit to a value below the observed stator current in Tuner. In the example below, the wheels began slipping at around 130 A.
+
+.. image:: images/slip-current.png
+   :alt: Wheel slip at 130 A stator current
+
+Preventing Brownouts
+^^^^^^^^^^^^^^^^^^^^
+
+The same strategy for improving battery life is applicable to brownouts as well. In the above example, we can see that our peak draw is 365 A. Brownouts occur when the robot voltage dips below a threshold (for the `FRC roboRIO <https://docs.wpilib.org/en/stable/docs/software/roborio-info/roborio-brownouts.html>`__, this threshold is around ~7 V). When the roboRIO dips below the threshold, it will disable all actuators to prevent a total robot reboot.
```

Comment:
This paragraph no longer has the context of the battery life section. We may want to pull some of that up here.

Note that at this point in time, everything has stator limits but not supply, so peak current is weird.